### PR TITLE
[now rm] Use the proper `client` so that the process does not hang

### LINKED ISSUE #2760: 404 error for removing deployment takes forever to exit
Try removing any deployment (might be related to having used an alias) and you will notice that the command (even thought it printed a 404) will take forever to exit:

### PR DESCRIPTION
**Pull request by TooTallNate**
The crux of this fix is that `getDeploymentByIdOrHost()` and `getProjectByIdOrName()` were improperly being passed the `Now` instance instead of the expected `Client` instance, and for some reason that would cause the process to hang until the underlying `http.Agent` timed out its connection to the API server.

Also ran `prettier` on this file.

Fixes #2760.

### CODE REVIEW

File: src/commands/remove.js [hunk 1]
```diff
@@ -131,43 +134,57 @@ export default async function main(ctx) {
...
+          !(d instanceof NowError) &&
+          (d.uid === id || d.name === id || d.url === normalizeURL(id))
+      );
 
     const [deploymentList, projectList] = await Promise.all([
-      Promise.all(ids.map((idOrHost) => getDeploymentByIdOrHost(now, contextName, idOrHost))),
-      Promise.all(ids.map(async (idOrName) => getProjectByIdOrName(now, idOrName)))
+      Promise.all(
+        ids.map(idOrHost =>
+          getDeploymentByIdOrHost(client, contextName, idOrHost)
```
Comment (TooTallNate) [inline src/commands/remove.js:148]: This, and line 152 below are the main fix

File: src/commands/remove.js [hunk 2]
```diff
@@ -131,43 +134,57 @@ export default async function main(ctx) {
...
+        ids.map(async idOrName => getProjectByIdOrName(client, idOrName))
+      )
     ]);
 
     deployments = deploymentList.filter(searchFilter);
     projects = projectList.filter(searchFilter);
 
     // When `--safe` is set we want to replace all projects
-    // with with deployments to verify the aliases
+    // with deployments to verify the aliases
```
Comment (TooTallNate) [inline src/commands/remove.js:160]: Fixed typo

File: src/commands/remove.js [hunk 3]
```diff
@@ -182,21 +199,24 @@ export default async function main(ctx) {
...
-      `or projects matching ` +
-      `${ids.map(id => chalk.bold(`"${id}"`)).join(', ')}. Run ${cmd('now ls')} to list.`
+        `or projects matching ` +
+        `${ids.map(id => chalk.bold(`"${id}"`)).join(', ')}. Run ${cmd(
+          'now ls'
+        )} to list.`
     );
-    return 0;
+    client.close();
+    return 1;
```
Comment (TooTallNate) [inline src/commands/remove.js:208]: Changed exit code to `1` to match Unix `rm does-not-exist`.

File: src/commands/remove.js [hunk 4]
```diff
@@ -237,23 +255,19 @@ export default async function main(ctx) {
     console.log(`${chalk.gray('-')} ${chalk.bold(project.name)}`);
   });
 
-  // if we close normally, we get a really odd error:
```
Comment (TooTallNate) [inline src/commands/remove.js:240]: Was able to remove this comment, because now this behavior does not happen.

File: src/commands/remove.js [hunk 5]
```diff
@@ -209,24 +229,22 @@ export default async function main(ctx) {
 
     if (confirmation !== 'y' && confirmation !== 'yes') {
       output.log('Aborted');
-      now.close();
-
-      // Hangs otherwise
-      process.exit(1);
```
Comment (TooTallNate) [inline src/commands/remove.js:215]: Removed comment and explicit `process.exit()` because they are no longer necessary

**Comment by styfle** [comment]
> Also ran prettier on this file.

Can you run prettier on every file in a new PR so we can stop this madness? 😆

File: src/commands/remove.js (hunk 1 above)
Comment (styfle) [reply src/commands/remove.js:148]: Is it possible to add a test for this? Or are there already tests for `remove.js`?

**Review by styfle — COMMENTED** [review]
I don't see any tests for `remove.js`

**Comment by TooTallNate** [comment]
Looks like you're right. I'll add some.

File: test/integration.js [hunk 6]
```diff
@@ -250,12 +280,22 @@ test('deploy a node microservice', async t => {
...
-  const response = await fetch(href);
+  let response = await fetch(href);
+  t.is(response.status, 200);
   const contentType = response.headers.get('content-type');
   const content = await response.json();
 
   t.is(contentType, 'application/json; charset=utf-8');
   t.is(content.id, contextName);
+
+  // Test that it can be deleted via `now rm`
```
Comment (TooTallNate) [inline test/integration.js:291]: New successful `now rm` test

File: test/integration.js [hunk 7]
```diff
@@ -1490,6 +1539,26 @@ test('print correct link in legacy warning', async t => {
   t.regex(stderr, /migrate-to-zeit-now/);
 });
 
+test('`now rm` 404 exits quickly', async t => {
```
Comment (TooTallNate) [inline test/integration.js:1542]: New `now rm` 404 test (specifically for this PR)

**Review by styfle — APPROVED** [review]
Thanks!

**Pull request merged**